sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs

```diff
@@ -86,6 +89,94 @@ public void DisposeDoesNotDisposesSharedPipeline()
             Assert.DoesNotThrow(() => transport.Client.CancelPendingRequests());
         }
 
+        [Test]
+        public async Task CookiesCanBeEnabledUsingSwitch()
+        {
+            using var appContextSwitch = new TestAppContextSwitch("Azure.Core.Pipeline.HttpClientTransport.EnableCookies", "true");
+            int requestCount = 0;
+            using (TestServer testServer = new TestServer(
+                context =>
+                {
+                    if (requestCount++ == 1)
+                    {
+                        Assert.IsTrue(context.Request.Headers.ContainsKey("cookie"));
+                        Assert.AreEqual(
+                            "stsservicecookie=estsfd",
+                            context.Request.Headers["cookie"].First());
+                    }
+
+                    context.Response.StatusCode = 200;
+                    context.Response.Headers.Add(
+                        "set-cookie",
+                        "stsservicecookie=estsfd; path=/; secure; samesite=none; httponly");
+                    return Task.CompletedTask;
+                },
+                https: true))
+            {
+                var transport = GetTransport(https: true);
+                Request request = transport.CreateRequest();
+                request.Method = RequestMethod.Post;
+                request.Uri.Reset(testServer.Address);
+                request.Content = RequestContent.Create("Hello");
+                var response = await ExecuteRequest(request, transport);
+
+                // create a second request to verify cookies not set
+                request = transport.CreateRequest();
+                request.Method = RequestMethod.Post;
+                request.Uri.Reset(testServer.Address);
+                request.Content = RequestContent.Create("Hello");
+                await ExecuteRequest(request, transport);
+            }
+        }
+
+        [Test]
+        public async Task CookiesCanBeEnabledUsingEnvVar()
+        {
+            string oldValue = Environment.GetEnvironmentVariable("AZURE_CORE_HTTPCLIENT_ENABLE_COOKIES");
+
+            Environment.SetEnvironmentVariable("AZURE_CORE_HTTPCLIENT_ENABLE_COOKIES", "true");
```

Comment:
use `TestEnvVar` https://github.com/Azure/azure-sdk-for-net/blob/e1f6bc5ad44470265f7b1df697173b57b2c8692a/sdk/core/Azure.Core.TestFramework/src/TestEnvVar.cs

Reply:
Ah I missed that.